### DISCUSSION TITLE
"Good" and "Bad" Notary Behavior

### DISCUSSION THREAD

**Discussion by panges2** [filecoin-project/notary-governance#802] [General]
## Context
With over 100 applications in the most recent Notary election, we suspect that there is great interest in joining the Notary community. As this community scales, we want to encourage “good” notary behavior and discourage “bad” notary behavior.
This discussion post aims to start a conversation and create a rough consensus on what constitutes "good" and "bad" notary behavior. This is important because, as the program advances, design decisions in notary tooling and incentive design will be based on these behavior metrics.  These notions are not meant to be hard metrics to single out notaries. Instead, they will give us some direction when designing hard metrics on a project-to-project basis. This post can also help new notaries understand their responsibilities.
## Notary Behavior
In the Fil+ governance team, we believe the following is a sufficient definition of “good” and “bad” behavior. By no means is this definition absolute. Feel free to comment below!
### Good Notary Behavior :+1::
- Due diligence before signing off the first DataCap tranche by getting answers to the following questions:
    - Is the data that the client claims to store worth the Fil+ subsidy?
    - Does the client have a sufficient level of credibility?
- Due diligence on a client befor signing off on subsequent DataCap tranches by getting answers to the following questions:
    - Does the client show any signs of DatacCap abuse? For example:
        - Is the client storing data that they initially claimed to be storing?
        - Are they storing data replicas properly with a large set of Storage providers?
- Performing due diligence in a timely manner, keeping a low time to DataCap.
- Engaging in the Fil+ community in accordance with their SLA level:
    - Joining bi-weekly Notary governance calls.
    - Contributing to discussions in the [notary governance repo](https://github.com/filecoin-project/notary-governance/discussions).
- Using Direct allocations as a method to test the credibility of clients.
- Reaching out to potential clients that qualify for the Fil+ subsidy.
### Bad Notary Behavior :-1::
- Being inactive as a notary.
    - Not signing any requests to issue DataCap.
    - Not performing public diligence on clients by asking questions.
    - Not engaging in the Fil+ governance community.
- Colluding with clients to abuse DataCap.
- Issuing DataCap to clients that are not following guidelines of the program or have been flagged by other notaries and are under further investigation.
## Related discussions:
https://github.com/filecoin-project/notary-governance/discussions/598
https://github.com/filecoin-project/notary-governance/discussions/230

**Comment by TimGuo7** [comment]
I think we can add some punishments for the bad behaviors of notaries here too.

**Reply by cryptowhizzard** [reply]
Stop gaslighting. https://filecoinproject.slack.com/archives/C01DLAPKDGX/p1673518708336059

**Comment by Dominic92** [comment]
Thanks, Mr. Pangestu for giving us the right way to do so.

The current cryptocurrency market is depressed regardless of whether it is primary or secondary. All investors will be excited by the increase in currency prices. 

If artificial deflation is needed to restore FIL's economic ecology to a virtuous circle, don't engage in such fearless quarrels and do something useful.

**Reply by cryptowhizzard** [reply]
Stop gaslighting. https://filecoinproject.slack.com/archives/C01DLAPKDGX/p1673518708336059

**Comment by qinlong2018** [comment]
Judging from the recent progress of Fil+, I think behaviors like @cryptowhizzard @herrehesse are "Bad" behaviors, and their existence is definitely not beneficial to ecological development.
I suggest removing all the positions of the above-mentioned personnel in the community so that the community can develop in a healthy way.

**Reply by cryptowhizzard** [reply]
Stop gaslighting. https://filecoinproject.slack.com/archives/C01DLAPKDGX/p1673518708336059

**Comment by herrehesse** [comment]
@qinlong2018 If you can't delete the message then just delete the messenger?
All we do is ask questions, which is why sometimes entities don't get what they want. No reason to remove anyone, ever.

You know what is also not beneficial for ecological development? Fraud by entities that abuse datacap for their own gain and store totally useless or even no data at all. That does not contribute to the enhancement of the value of the Filecoin network.

A few notaries asking questions is the problem according to you. Strange.

**Reply by 001can** [reply]
@herrehesse “raud by entities that abuse datacap for their own gain and store totally useless or even no data at all”

**Reply by cryptowhizzard** [reply]
Stop gaslighting. https://filecoinproject.slack.com/archives/C01DLAPKDGX/p1673518708336059

**Reply by 001can** [reply]
that seems to be exactly what you guys are doing

**Reply by qinlong2018** [reply]
@cryptowhizzard @herrehesse What a hegemonic thought, if you hear a voice that dissatisfies you, you want to delete it! Maybe you don't even realize how arrogant and prejudiced you are!
Didn't the LDN be questioned and scrutinized by the notary before this period of your "going crazy"? You are denying all the work of other notaries!
and you
You maliciously suppress LDN in Asia, especially China, and ask endless questions. This is your regional discrimination, which is intolerable!
In addition, you collude with small groups to sign with each other, apply for a large number of repeated worthless LDNs, and sell them for profit.
During the LDN review, you repeatedly attacked the notary and the client and maliciously framed them.
You are holding the banner of rules and justice, but the motive behind it is dirty and dirty.
You want to eat by yourself, but break other people's jobs.

We are continuing to investigate your various evil deeds, I will create a new discussion to expose your malicious behavior in FIL+, wait and see!

**Comment by alchemypunk** [comment]
## "Good" and "Bad" notary behavior is not the point
Get to the point of the question by reading what is being discussed.
I have found that notaries only need a standard version of due diligence, and that this standard set of due diligence can be continuously improved and iterated on.
This due diligence can be generalized and summarized from the current general version.

### :dart: :dart: :dart: The benefits of doing this :dart: :dart: :dart:
1. The community has perfected a set of SOP standards, which can prevent each organization from fighting independently;
2. The community can use resources and strength to jointly face real and demanding big data storage customers;
3. Less controversy, more focus on real storage.

### :bell: :bell: :bell:Also suggest :bell: :bell: :bell:
1. More outstanding notaries share their strategies and market promotion plans (How do they impress customers to do real storage? How do they do their due diligence?), and can even require TOP 5% of notaries to do so. I believe that if notaries are doing real work, then their sharing must be true.
2. Set the red line level (punishment level), which is divided according to the amount of data. Because sometimes the fault is not with the notary, but with the fake LDN applicant :japanese_goblin:.

**Reply by cryptowhizzard** [reply]
That sounds like a plan.

Perhaps working in teams is also something we can explore. A team should consist of members who can do technical, operational and commercial due diligence 

I read remarks that some notary's are not capable of 

- Doing retrieval testing
- Doing testing if a SP is online and understand the errors when a SP is offline.
- Understanding CID's and what it means if they find a duplicate.
- Doing proper and real KYC. 
- Evaluate the data that is stored and test / see if it is really the data an applicant indicated to store.

This sounds like an alarm bell that the structure and coordination in the organization is not optimal. We should use the talents where available.

**Reply by kevzak** [reply]
I agree @0xXPunkX @cryptowhizzard - we should look to move away from any subjective measure of due diligence towards an SOP with more clear and concise categories or metrics to evaluate for every application.

**Reply by panges2** [reply]
@0xXPunkX I am in no way opposed to this idea. I plan to do follow-up posts to this one detailing harder metrics/SOPs to keep notaries accountable. 

This discussion post was in no way supposed to be the end-all be-all. Instead, I wanted this discussion post to be a basis that reinforces our decisions when making the harder metrics. My logic goes: to agree on hard metrics/SOPs without going back and forth, we first need to agree on the general role of a notary. (This post might be common sense to you, but others may not agree with this) At the same time, we need to agree on this before deciding how to design other things too, such as accountability tooling/penalties/incentives. 

Thank you for the suggestions. I love that you're thinking ahead!

**Reply by panges2** [reply]
@cryptowhizzard, You bring up a great point here. I also believe that most notaries are incapable of doing the technical tasks on that list (1,2,5). However, for 3 and 4, I think that it is just a matter of establishing KYC standards as well as proper training. 

1,2,5 are very necessary as they are the most objective way of doing due diligence on the client. At the end of the day, it does not really matter what the client says if the data that they are providing is empty/fraudulent. But now this raises a new question about the role of the notary: 

as we advance, should these technical tasks be a requirement for the notary (option 1), or should it be a role someone more downstream should be responsible for (option 2)?

Option 1 could be done in a few ways:
- First, your suggestion: A Notary should not be an individual. Instead, it should be a team (notary team) with one technical, operational, and commercial due diligence. This would be ideal; however, enforcing this would not be easy. I don't know how many notaries out there have a technical counterpart. Furthermore, I'm not sure how many people in the ecosystem today can do these technical tasks such as retrieval testing. 
- Second idea: We could require notaries to have to be able to do these technical things. This could be done by training notaries who apply or requiring them to be able to do them before ever applying. The problem is how many notaries are willing to spend that much time learning these technical skills.

Option 2: Technical due diligence should be someone else's responsibility, while notaries should be responsible for more top-level due diligence. We could enforce a rule to make data preparers/ SPs to do checks on the data being stored. This is a whole other topic that can be discussed.

Option 3: I'm open to hearing anyone's thoughts on this if there are other ways around this. 

Thank you for raising these points and adding to the discussion!

**Comment by stcloudlisa** [comment]
Glad to be able to join the discussion here.

We need to formulate a rule and set a fault tolerance range:

1. What kind of mistakes must not happen. Once mistakes are made, all notaries should not sign for them.

2. Since the emergence of bots, we have seen many deceptions, how to deal with these deceptions.

3. Which errors are relatively minor, and whether it is possible to give the customer a chance to correct.

4. How do we identify customers, whether we trust customers, and how to trust customers.

**Reply by alchemypunk** [reply]
A fault tolerance range is very importain, Guiding and improving is more valuable than finger pointing.
Anyone can find a problem, but solving it is a difficult habit.

**Reply by herrehesse** [reply]
Everyone deserves a second chance, but there should be zero tolerance for bad actors. We should not have to be kind or friendly to anyone trying to abuse the system. It hurts everyone.

Comply, follow the rules, be transparent. We will update the bots until good and bad behavior is visible and act on it promptly. Again, everyone deserves a second chance.

**Reply by kevzak** [reply]
Posting here @panges2 as an idea for discussion related to good and bad behavior. I really think we need to start separating initial application/allocation review vs. subsequent reviews. The first should be focused on client/business/data KYC and storage plan review and the rest should be focused on usage of data cap (did they follow the plan they set)

Some notaries have openly said they are not "technical" enough to do retrieval checks or understand how to know how DataCap was used. Then they should not review 2+ allocation checks, IMO. 

Perhaps this highlights notary categorization? or levels? I don't think all notaries should have the same level of approval from the start. It comes with experience, training, reputation, etc.

**Reply by panges2** [reply]
@kevzak I agree. The question remains whether we want all notaries to be technical or if there should be another role/tier for retrieval testing on subsequent allocations. I think Im more with the former, but for this to happen, we need to make tooling ("validation bot, CID checker, notary leaderboard, canvas") that makes it easy for them to do checks. This is the direction we're heading.

**Reply by liyunzhi-666** [reply]
@panges2 I don't think we should require all notaries to have a technical background.
I've been following the FiL+ program since V1 Notary.
I remember the project started with the vision that we wanted notaries to come from different industries and fields. Currently it takes some time and technology for notaries to retrieve documents when doing due diligence. So shouldn't we provide more tools to assist notaries?

**Reply by cryptowhizzard** [reply]
I want to add here, detecting “things” requires knowledge. For example cid share requires ipfs knowledge to understand it’s fundamentals when it is wrong or right.

On the other end, commercial factors should also be at play. Most people are either A or B leaving us with either toolings in combi with thorough explanations or grouping.
